### PR TITLE
Ping mxnet nightly to the lastest path

### PR DESCRIPTION
**Pull request by apeforest**
MXNet is now publishing nightly to a webpage. Update tests to the correct path.

@karan6181 @eric-haibin-lin please review.

### CODE REVIEW

File: Dockerfile.test.cpu [hunk 1]
```diff
@@ -113,7 +113,11 @@ RUN if [[ ${PYTORCH_PACKAGE} == "torch-nightly" ]]; then \
 RUN pip install "Pillow<7.0" --no-deps
 
 # Install MXNet.
-RUN pip install ${MXNET_PACKAGE}
+RUN if [[ ${MXNET_PACKAGE} == "mxnet-nightly" ]]; then \
+        pip install -pre mxnet -f https://dist.mxnet.io/python/all; \
+    else \
+        pip install ${MXNET_PACKAGE} \
```
Comment (tgaddair) [inline Dockerfile.test.cpu]: I think you're missing a `;` here.

Comment (apeforest) [reply Dockerfile.test.cpu]: Thanks for the good catch!

File: Dockerfile.test.cpu [hunk 2]
```diff
@@ -113,7 +113,11 @@ RUN if [[ ${PYTORCH_PACKAGE} == "torch-nightly" ]]; then \
 RUN pip install "Pillow<7.0" --no-deps
 
 # Install MXNet.
-RUN pip install ${MXNET_PACKAGE}
+RUN if [[ ${MXNET_PACKAGE} == "mxnet-nightly" ]]; then \
+        pip install -pre mxnet -f https://dist.mxnet.io/python/all; \
```
Comment (tgaddair) [inline Dockerfile.test.cpu]: On this line, it should be `--pre` instead of `-pre`.

File: Dockerfile.test.gpu [hunk 3]
```diff
@@ -96,7 +96,11 @@ RUN if [[ ${PYTORCH_PACKAGE} == "torch-nightly" ]]; then \
 RUN pip install "Pillow<7.0" --no-deps
 
 # Install MXNet.
-RUN pip install ${MXNET_PACKAGE}
+RUN if [[ ${MXNET_PACKAGE} == "mxnet-nightly" ]]; then \
+        pip install -pre mxnet_cu100 -f https://dist.mxnet.io/python/all; \
```
Comment (tgaddair) [inline Dockerfile.test.gpu]: Tests for the nightly builds are running on CUDA 10.1.  Should this be `mxnet_cu101`?

File: Dockerfile.test.cpu (hunk 2 above)
Comment (apeforest) [reply Dockerfile.test.cpu]: Fixed.

File: Dockerfile.test.gpu (hunk 3 above)
Comment (apeforest) [reply Dockerfile.test.gpu]: Updated to cu101

**Comment by tgaddair** [comment]
Hey @apeforest, looks like MXNet nightly is causing tests to fail: https://buildkite.com/horovod/horovod/builds/2116

Can you investigate?

**Comment by apeforest** [comment]
Looking into it

**Comment by apeforest** [comment]
I don't find issue when I build horovod locally with mxnet nightly. One possible issue is that MXNet stopped supporting python 2.7 since the latest release. I will make changes to the build script.

**Comment by apeforest** [comment]
@tgaddair Created another PR https://github.com/horovod/horovod/pull/1746 to skip mxnet nightly when Python version is 2.7

**Pull request merged**